### ISSUE TITLE
enumerate adapters

### ISSUE THREAD

**Issue by toloudis** [pygfx/wgpu-py#186]
Would it be possible to enumerate adapters for offscreen rendering / multi-gpu tasks?  I'm not sure this functionality is exposed currently.

**Comment by almarklein** [comment]
When you request an adapter you can provide a power-preference, and force a backend (Vulkan/Metal/DX12), but that's it. The wgpu API does not provide (yet?) a way to list available adapters.

**Comment by toloudis** [comment]
When I saw this code (linked below), that was what made me wonder.  It would be great if we could use wgpu to use multiple adapters/devices or even just have more control to select adapter of choice.
https://github.com/gfx-rs/wgpu/blob/647f6ee9f2c479c1c6469841445d4d6cfb10c96a/wgpu-core/src/instance.rs#L509

**Comment by toloudis** [comment]
https://github.com/gfx-rs/wgpu/discussions/2022

**Comment by almarklein** [comment]
> When I saw this code (linked below), that was what made me wonder.

Yeah, I know it would be nice, but possibly also a security hole, so might not be a part of WebGPU. Might be exposed via wgpu-native, but not sure if that's the plan.

Anyway, you can hack it a bit like this:
```
import wgpu.backends.rs

adapter1 = wgpu.request_adapter(canvas=None, power_preference="low-power")
adapter2 = wgpu.request_adapter(canvas=None, power_preference="high-performance")

if adapter1.properties["deviceID"] == adapter2.properties["deviceID"]:
    pass  # there is one adapter
else:
    pass  # at least two adapters
```
